src/main/java/net/dv8tion/jda/core/utils/WidgetUtil.java

```diff
@@ -0,0 +1,685 @@
+     */
+    public static String getWidgetBanner(String guildId, BannerType type)
+    {
+        Args.notNull(guildId, "GuildId");
+        Args.notNull(type, "BannerType");
+        return String.format(WIDGET_PNG, guildId, type.name().toLowerCase());
+    }
+    
+    /**
+     * Represents the available banner types
+     * Each of these has a different appearance:
+     * 
+     * Shield - tiny, only contains Discord logo and online count
+     * 
+     * Banner1 - medium, contains server name, icon, and online count, and a 
+     * "Powered by Discord" bar on the bottom
+     * 
+     * Banner2 - small, contains server name, icon, and online count, and a 
+     * Discord logo on the side
+     * 
+     * Banner3 - medium, contains server name, icon, and online count, and a 
+     * Discord logo with a "Chat Now" bar on the bottom
+     * 
+     * Banner4 - large, contains a very big Discord logo, server name, icon, and 
+     * online count, and a big "Join My Server" button
+     */
+    public enum BannerType
+    {
+        SHIELD, BANNER1, BANNER2, BANNER3, BANNER4
+    }
+    
+    
+    /**
+     * Gets the pre-made HTML Widget for the specified guild using the specified
+     * settings. The widget will only display correctly if the guild in question
+     * has the Widget enabled.
+     * 
+     * @param guild the guild
+     * @param theme the theme, light or dark
+     * @param width the width of the widget
+     * @param height the height of the widget
+     * @return a String containing the pre-made widget with the supplied settings
+     */
+    public static String getPremadeWidgetHtml(Guild guild, WidgetTheme theme, int width, int height)
+    {
+        Args.notNull(guild, "Guild");
+        return getPremadeWidgetHtml(guild.getId(), theme, width, height);
+    }
+    
+    /**
+     * Gets the pre-made HTML Widget for the specified guild using the specified
+     * settings. The widget will only display correctly if the guild in question
+     * has the Widget enabled. Additionally, this method can be used independently
+     * of being on the guild in question.
+     * 
+     * @param guildId the guild ID
+     * @param theme the theme, light or dark
+     * @param width the width of the widget
+     * @param height the height of the widget
+     * @return a String containing the pre-made widget with the supplied settings
+     */
+    public static String getPremadeWidgetHtml(String guildId, WidgetTheme theme, int width, int height)
+    {
+        Args.notNull(guildId, "GuildId");
+        Args.notNull(theme, "WidgetTheme");
+        Args.notNegative(width, "Width");
+        Args.notNegative(height, "Height");
+        return String.format(WIDGET_HTML, guildId, theme.name().toLowerCase(), width, height);
+    }
+    
+    /**
+     * Represents the color scheme of the widget
+     * 
+     * These color themes match Discord's dark and light themes
+     */
+    public enum WidgetTheme
```

Comment:
By this i mean the 2 enums provided in this class should be in next to each other. Just below the methods, just above the internal classes (Member/Channel)